common.go

```diff
@@ -150,3 +150,66 @@ const (
 	KeyUpdateNotRequested KeyUpdateRequest = 0
 	KeyUpdateRequested    KeyUpdateRequest = 1
 )
+
+type State uint8
```

Comment:
Adding this enum is a good idea, but should be in a separate PR.  When things are easily decoupled, I would much rather review a small PR than a large one.

Reply:
#151.